apps/src/code-studio/progress.js

```diff
@@ -302,13 +304,16 @@ function initializeStoreWithProgress(
 
   store.dispatch(setIsAge13Required(scriptData.age_13_required));
 
-  // Progress from the server should be written down locally, unless we're a teacher
-  // viewing a student's work.
+  // Write progress locally unless we're a teacher viewing student work
   if (!isViewingStudentAnswer) {
```

Comment:
NIT: isn't `!isViewingStudentAnswer` the general case? like, usually this code is being called when viewing one's own progress? it seems to me like merging progress and subscribing to progress updates are the bulk of the work in this function, so rather than putting them in a conditional block i'd consider replacing this with `if (isViewingStudentAnswer) { return }` (maybe at line 290 actually, to be super clear).

imo that would make this a little easier to read. i.e. "okay, if we're a teacher viewing work, we're done here".

Reply:
That makes a lot more sense. I'll update it to match that. Thanks you!!

Reply:
👍